### PR TITLE
[14.0] Add connector_importer_product

### PR DESCRIPTION
**Pull request by mmequignon**
Base product importer.

### CODE REVIEW

File: connector_importer_product/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Generic module to import product throught SFTP
```
Comment (simahawk) [inline connector_importer_product/readme/DESCRIPTION.rst]: is not for SFTP only

File: connector_importer_product/data/import_recordset.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2022 Camptocamp SA
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+  <record id="import_recordset_product_category" model="import.recordset">
+    <field name="sequence" eval="20" />
+    <field name="backend_id" ref="import_backend" />
+    <field name="source_model">import.source.csv.sftp</field>
```
Comment (sebalix) [inline connector_importer_product/data/import_recordset.xml]: Should not be tied to SFTP only
```suggestion
    <field name="source_model">import.source.csv.sftp</field>
```

File: connector_importer_product/data/import_source.xml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+  <record id="import_source_csv_product_category" model="import.source.csv.sftp">
+    <field name="name">product_category</field>
+    <field name="chunk_size" eval="0" />  <!-- do not chunk CSV lines -->
+    <field name="csv_delimiter">,</field>
+    <field name="csv_encoding">utf-8</field>
+    <field name="storage_id" ref="storage_backend_importer" />
+    <field name="move_file_after_import" eval="1" />
```
Comment (sebalix) [inline connector_importer_product/data/import_source.xml]: This is also a configuration brought by SFTP, to remove I guess

File: connector_importer_product/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    ],
+    "data": [
+        # storage_backend
+        "data/storage_backend.xml",
+        # Imports to initialize data for production
+        "data/import_backend.xml",
+        "data/import_type.xml",
+        "data/import_source.xml",
+        "data/import_recordset.xml",
+        # Demo
```
Comment (TDu) [inline connector_importer_product/__manifest__.py]:
```suggestion
```

File: connector_importer_product/components/common.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class TranslationImporter(Component):
+    _name = "translation.importer"
+    _inherit = "importer.record"
+
+    _break_on_error = True
+
+    def make_translation_key(self, key, lang):
+        """Overridden to handle 'name_fr' and 'name_de' fields."""
```
Comment (TDu) [inline connector_importer_product/components/common.py]: Need to be reformulated it is not specific to fr or de, explain its use.

File: connector_importer_product/components/product_attribute.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.addons.component.core import Component
+from odoo.addons.connector.components.mapper import mapping
+
+from ..utils import sanitize_external_id
+
+
+class ProductAttributeRecordImporter(Component):
```
Comment (TDu) [inline connector_importer_product/components/product_attribute.py:10]:
```suggestion
class ProductAttributeImporter(Component):
```

File: connector_importer_product/components/product_attribute_value.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.addons.component.core import Component
+from odoo.addons.connector.components.mapper import mapping
+from odoo.addons.connector_importer.utils.mapper_utils import xmlid_to_rel
+
+from ..utils import sanitize_external_id
+
+
+class ProductAttributeValueRecordImporter(Component):
```
Comment (TDu) [inline connector_importer_product/components/product_attribute_value.py:10]:
```suggestion
class ProductAttributeValueImporter(Component):
```

File: connector_importer_product/components/product_category.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                    for x in value.split(",")
+                    if self.env.ref(x, raise_if_not_found=False)
+                ],
+            )
+        ]
+
+    return modifier
+
+
+class ProductCategoryRecordImporter(Component):
```
Comment (TDu) [inline connector_importer_product/components/product_category.py:34]:
```suggestion
class ProductCategoryImporter(Component):
```

File: connector_importer_product/components/product_packaging.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.addons.component.core import Component
+
+from ..utils import sanitize_external_id
+
+
+class ProductPackagingRecordImporter(Component):
```
Comment (TDu) [inline connector_importer_product/components/product_packaging.py:9]:
```suggestion
class ProductPackagingImporter(Component):
```

**Review by TDu — COMMENTED** [review]
About my comment of the class name I just stopped after a while of reviewing, I means it is not that important

File: connector_importer_product/data/import_backend.xml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2022 Camptocamp SA
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+  <record id="import_backend" model="import.backend">
```
Comment (simahawk) [inline connector_importer_product/data/import_backend.xml]: you should trash all these records

File: connector_importer_product/__manifest__.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    "name": "Connector Importer SFTP Product",
+    "summary": "Product Importer",
+    "version": "14.0.1.0.0",
+    "category": "Tools",
+    "website": "https://github.com/OCA/connector-interfaces",
+    "author": "Camptocamp SA, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "depends": [
+        # oca
+        "connector_importer_source_sftp",
```
Comment (simahawk) [inline connector_importer_product/__manifest__.py]: this should not be a dependency

File: connector_importer_product/__manifest__.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        # src
+        "product",
+    ],
+    "data": [
+        # storage_backend
+        "data/storage_backend.xml",
```
Comment (simahawk) [inline connector_importer_product/__manifest__.py]: same

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by simahawk** [comment]
@sebalix @mmequignon I've pushed some cleanup/fixes/improvements.

**Review by mmequignon — COMMENTED** [review]
@simahawk I cannot approve my own pr, but lg.
Thx!

**Comment by simahawk** [comment]
I switched tests to GH actions on #89.Rebased and squashed here.

**Review by simahawk — APPROVED** [review]

**Pull request merged**